File: pqueue/heapsort.cpp
#include <iostream>
#include "heap.h"
using namespace std;

int main(){
	int n;cin>>n;
	vector<int> v(n);
	for(int i=0;i<n;++i)cin>>v[i];
	heap<int> h(v);
	while(h.size()){
		cout<<h.top()<<' ';
		h.pop();
	}
	cout<<'\n';
}

File: pqueue/heapsort.py
from heap import Heap
import random

def heapsort(arr):
    heap = Heap(0.5, 0.2)
    for i in arr:
        heap.push(i)
    ret = []
    while not heap.empty():
        ret.append(heap.top())
        heap.pop()
    return ret

a = [i for i in range(1, 9)]

random.shuffle(a)

b = heapsort(a)

print("a: " + str(a))
print("b: " + str(b))
